conans/test/functional/profiles/profiles_test.py

```diff
@@ -0,0 +1,105 @@
+import textwrap
+import unittest
+
+from conans.client.tools import environment_append
+from conans.test.utils.tools import TestClient
+
+from configparser import ConfigParser, BasicInterpolation
+
+
+class ProfileInterpolator(BasicInterpolation):
+    """Interpolation which expands environment variables in values."""
+
+    def before_get(self, parser, section, option, value, defaults):
+        from collections import defaultdict
+        rawval = parser.get(section, option, raw=True)
+        missing_keys = defaultdict(str)
+        rawval % missing_keys
+        all_keys = dict(missing_keys)
+        if section in parser.section_defaults:
+            all_keys.update(parser.section_defaults[section])
+        val = rawval % all_keys
+        return val
+
+
+class ProfileConfigParser(ConfigParser):
+
+    def __init__(self, section_defaults=None, *args, **kargs):
+        super(ConfigParser, self).__init__(interpolation=ProfileInterpolator(), delimiters=("=",),
+                                           *args, **kargs)
+        self.section_defaults = section_defaults or {}
+
+    def optionxform(self, optionstr):
+        return optionstr
+
+    def items(self, *args, **kargs):  # section=_UNSET, raw=False, vars=None
+        items = dict(super(ConfigParser, self).items(*args, **kargs))
+        result = {}
+        section = args[0] if args else None
+        if section:
+            if section in self.section_defaults:
+                result = self.section_defaults[section]
+        result.update(items)
+        return result
+
+
+class ProfileTest(unittest.TestCase):
+
+    def test_env(self):
```

Comment:
I am trying to have a test passing like:

```python
    def test_os_environ(self):
        with environment_append({"CXXFLAGS": "-fPIC"}):
            parser2 = ProfileConfigParser()
            profile2 = textwrap.dedent("""
                [env]
                foo=foo1
                bar=bar2
                CXXFLAGS=%(CXXFLAGS)s -fother
                OTHERFLAGS=%(missing_flag)s -fother
            """)
            parser2.read_string(profile2)
            self.assertDictEqual({"foo": "foo1", "CXXFLAGS": "-fPIC -fother", "bar": "bar2",
                                  "zet": "zet", "OTHERFLAGS": " -fother"},
                                 dict(parser2.items("env")))
```
But not evident. Env-vars should be readable from the environment too, not only from other profiles. Is this possible?

Reply:
Not in this POC, as the main issue was to create a substitution syntax from one profile to another. However, you could do the env var substitution by means of `os.path.expandvars` or similar the same way it is done the interpolation here with values from other profiles